### PR TITLE
✨ [FEATURE] book actions (ADD-REMOVE-UPDATE)

### PR DESCRIPTION
**Pull request by ronsigter**
### Changes
- [x] Adds hooks to add-remove-update `reading_list` of a book
- [x] Adds `reading_list_id` in book's serializer filtered by the current user 

### Demo
add-update-remove

### CODE REVIEW

File: backend/app/serializers/book_serializer.rb [hunk 1]
```diff
@@ -31,4 +31,8 @@ class BookSerializer
   attribute :reading_status do |book, params|
     book.reading_lists.find_by(user_id: params[:current_user]&.id)&.status
   end
+
+  attribute :reading_list_id do |book, params|
+    book.reading_lists.find_by(user_id: params[:current_user]&.id)&.id
+  end
```
Comment (ronsigter) [inline backend/app/serializers/book_serializer.rb:37]: Needed the `reading_list.id` here in order for me to update the status and/or delete it. 

FE will send the request to BE `/api/v1/reading_lists/:id` 

Lemme know a better implementation for this 👍

File: frontend/components/cards/BookCard.tsx [hunk 2]
```diff
@@ -1,30 +1,54 @@
+'use client'
```
Comment (jimryanzulueta) [inline frontend/components/cards/BookCard.tsx:1]: Is this intended?

File: frontend/hooks/useReadingListActions.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  const { data: session } = useSession()
+  const queryClient = useQueryClient()
+
+  const add = useMutation({
+    mutationFn: async (vars: { book_id: string }) => {
+      const { data } = await addToReadingList(session, vars)
+      return data
+    },
+    onSuccess: () => queryClient.invalidateQueries(['books'])
+  })
```
Comment (jimryanzulueta) [inline frontend/hooks/useReadingListActions.ts:20]: Oh cool. Haven't used `useMutation` this much from `react-query`

**Review by jimryanzulueta — APPROVED** [review]
Awesome <3 Looks good~

File: backend/app/serializers/book_serializer.rb (hunk 1 above)
Comment (jimryanzulueta) [reply backend/app/serializers/book_serializer.rb:37]: I'd personally do the same so this is fine with me.

File: frontend/components/cards/BookCard.tsx (hunk 2 above)
Comment (ronsigter) [reply frontend/components/cards/BookCard.tsx:1]: Nawp. It's a [convention](https://beta.nextjs.org/docs/rendering/server-and-client-components#convention) for the NextJSv13 app directory to use client side rendering. 😄

Comment (jimryanzulueta) [reply frontend/components/cards/BookCard.tsx:1]: Oooh. Haven't upgraded yet. But good to know!

**Pull request merged**